test/ModuleLoader.test.cpp

```diff
@@ -1,8 +1,20 @@
 #include "GameModules/IpDropDown.h"
+#include "GameModules/IniModule.h"
 #include "ModuleLoader.h"
 #include <gtest/gtest.h>
 #include <memory>
+#include <stdexcept>
 
+// Use to test that two modules with the same name but different casing is rejected from registering with the ModuleLoader
+// Windows file system and .ini key value pairs are case insensitive
+class DifferentCasedNameModule : public GameModule
+{
+public:
+	DifferentCasedNameModule() : GameModule("ipdropdown") {}
```

Comment:
It may make more sense to pass the name in as a constructor parameter. Then you can create two instances of the test class with the same name but different cases. That way you're not forced to match up with some real module that gets borrowed for this test case and has no real relation to it.

```
Class DummyTestModule : public GameModule
{
public:
    DummyTestModule(const char* moduleName) : GameModule(moduleName) {}

    void Load() override {}
    bool Unload() override { return true; }
};

// ... 
// Names are case insensitive and must be unique
moduleManager.RegisterModule(DummyTestModule("DummyTestModule"));
moduleManager.RegisterModule(DummyTestModule("dummytestmodule")); // Error
```